ARCHITECTURE.md

```diff
@@ -1,113 +1,265 @@
+
+[Server]() and [Client]() ActionPlayers are companion classes to actions that are used to actually play out the actions on both client and server
+
+#### Movement action flow
+
+- Client clicks mouse on target destination.
+- Client->server RPC, containing target destination.
+- Anticipatory animation plays immediately on client.
+- Server performs pathfinding.
+- Once pathfinding is finished, server representation of entity starts updating its NetworkVariables at the same cadence as FixedUpdate.
+- Visuals GameObject never outpaces the simulation GameObject, and so is always slightly behind and interpolating towards the networked position and rotation.
+
+### Navigation system
+
+Each scene which uses navigation or dynamic navigation objects should have a `NavigationSystem` component on a scene GameObject. That object also needs to have the `NavigationSystem` tag.
+
+#### Building a navigation mesh
+
+The project is using `NavMeshComponents`. This means direct building from the Navigation window will not give the desired results. Instead find a `NavMeshComponent` in the given scene e.g. a __NavMeshSurface__ and use the __Bake__ button of that script. Also make sure that there is always only one navmesh file per scene. Navmesh files are stored in a folder with the same name as the corresponding scene. You can recognize them based on their icon in the editor. They follow the naming pattern "NavMesh-<name-of-creating-object\.asset>"
+
+## Important architectural patterns and decisions
+
+In Boss Room we made several noteworthy architectural decisions:
+
+### Dependency Injection
+
+We use [Dependency Injection](https://en.wikipedia.org/wiki/Dependency_injection) pattern, with our library of choice being [VContainer](https://vcontainer.hadashikick.jp/).
+
+DI allows us to clearly define our dependencies in code, as opposed to using static access, pervasive singletons or scriptable object references (aka Scriptable Object Architecture). Code is easy to version-control and comparatively easy to understand for a programmer, as opposed to Unity YAML-based objects, such as scenes, scriptable object instances and prefabs.
+
+DI also allows us to circumvent the problem of cross-scene references to common dependencies, even though we still have to manage the lifecycle of MonoBehaviour-based dependencies by marking them with DontDestroyOnLoad and destroying them manually when appropriate.
+
+### Client/Server code separation
+
+Original problem: reading code with client and server code mixing together adds complexity and makes it easier to make mistakes. That code will often run in a single context, either client or server.
+
+Boss Room explored different client-server code separation approaches. We decided in the end to revert our initial client/server/shared assemblies to a more classic domain driven assembly architecture, but still keeping certain more complex classes with a client/server separation.
```

Comment:
```suggestion
Boss Room explored different client-server code separation approaches. For readers that have been following us since the beginning, we decided in the end to revert our initial client/server/shared assemblies to a more classic domain driven assembly architecture, but still keeping certain more complex classes with a client/server separation.
```